mmv1/third_party/terraform/services/sql/resource_sql_database_instance.go.tmpl

```diff
@@ -2504,6 +2508,8 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	// If there is no change detected in disk size, no need to try and update the disk size, send 0 always
+	instance.Settings.DataDiskSizeGb = 0
```

Comment:
this seems like a non-op change, isn't instance immediately getting overwritten with a new object in the next line?

Reply:
Also, I am not confident about eroding the disk size. Why not set it to what the current value is instead of always zero? Are you sure this is safe?